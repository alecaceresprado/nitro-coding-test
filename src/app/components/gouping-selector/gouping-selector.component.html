<div class="grouping-selector">
  <div class="grouping-selector__title">Options</div>
  <div class="grouping-selector__criteria">
    Order Criteria
    <p
      *ngFor="let criteriaElement of criteria"
      class="grouping-selector__criteria-element"
      [class.grouping-selector__criteria-element--selected]="
        criteriaElement === selectedCriteria
      "
      (click)="handleCriteriaChange.emit(criteriaElement)"
    >
      {{ criteriaElement }}
    </p>
  </div>
</div>
