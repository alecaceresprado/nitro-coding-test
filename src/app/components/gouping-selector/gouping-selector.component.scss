.grouping-selector {
  &__title {
    text-align: center;
    background: lightgray;
    height: 2rem;
    padding-top: 1rem;
    margin-bottom: 1rem;
  }

  &__criteria {
    padding: 0 .5rem;
  }

  &__criteria-element {
    margin: .25rem;
    padding-left: 1rem;

    &--selected {
      background: lightblue;
    }
  }
}
