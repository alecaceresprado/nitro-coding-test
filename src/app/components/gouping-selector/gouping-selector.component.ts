import { Component, EventEmitter, Output, Input } from '@angular/core';

import { OrderCriteria } from '@models';

@Component({
  selector: 'app-gouping-selector',
  templateUrl: './gouping-selector.component.html',
  styleUrls: ['./gouping-selector.component.scss'],
})
export class GoupingSelectorComponent {
  @Input() public selectedCriteria: OrderCriteria = OrderCriteria.WEEK;
  @Output() public handleCriteriaChange: EventEmitter<
    OrderCriteria
  > = new EventEmitter();

  public criteria: OrderCriteria[] = [
    OrderCriteria.WEEK,
    OrderCriteria.LOCATION,
    OrderCriteria.AUTHOR,
  ];
}
