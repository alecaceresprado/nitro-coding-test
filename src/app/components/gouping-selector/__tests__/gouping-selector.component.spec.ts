import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { GoupingSelectorComponent } from '../gouping-selector.component';

describe('GoupingSelectorComponent', () => {
  let component: GoupingSelectorComponent;
  let fixture: ComponentFixture<GoupingSelectorComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [GoupingSelectorComponent],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(GoupingSelectorComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should create', () => {
    expect(fixture).toMatchSnapshot();
  });
});
