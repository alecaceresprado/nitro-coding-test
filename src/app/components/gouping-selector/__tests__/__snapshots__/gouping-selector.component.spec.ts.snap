// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoupingSelectorComponent should create 1`] = `
<app-gouping-selector
  criteria={[Function Array]}
  handleCriteriaChange={[Function EventEmitter]}
  selectedCriteria={[Function String]}
>
  <div
    class="grouping-selector"
  >
    <div
      class="grouping-selector__title"
    >
      Options
    </div>
    <div
      class="grouping-selector__criteria"
    >
       Order Criteria 
      
      <p
        class="grouping-selector__criteria-element grouping-selector__criteria-element--selected"
      >
         By posting week 
      </p>
      <p
        class="grouping-selector__criteria-element"
      >
         By location 
      </p>
      <p
        class="grouping-selector__criteria-element"
      >
         By author 
      </p>
    </div>
  </div>
</app-gouping-selector>
`;
